homeassistant/components/recollect_waste/sensor.py

```diff
@@ -82,25 +76,19 @@ def __init__(
     @callback
     def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
-        if self.entity_description.key == SENSOR_TYPE_CURRENT_PICKUP:
-            try:
-                event = self.coordinator.data[0]
-            except IndexError:
-                LOGGER.error("No current pickup found")
-                return
+        relevant_events = (e for e in self.coordinator.data if e.date >= date.today())
+        pickup_index = self.PICKUP_INDEX_MAP[self.entity_description.key]
+
+        try:
+            for _ in range(pickup_index):
+                event = next(relevant_events)
+        except StopIteration:
+            LOGGER.info("No pickup event found for %s", self.entity_description.key)
```

Comment:
Is this really a user facing log message? What should the user do with this information? If it's a developer message please lower to debug level.

Reply:
https://github.com/home-assistant/core/pull/85766